imports/plugins/core/checkout/client/templates/checkout/login/login.js

```diff
@@ -17,10 +16,9 @@ Template.checkoutLogin.helpers({
     const cart = Cart.findOne();
     if (cart && cart.workflow) {
       const currentStatus = cart.workflow.status;
-      const guestUser = Reaction.hasPermission("guest", Meteor.user());
-      const anonUser = Roles.userIsInRole("anonymous", Meteor.user(), Reaction.getShopId());
+      const guestUser = Reaction.hasPermission("guest", Meteor.userId(), Reaction.getShopId());
```

Comment:
Don't need Meteor.userId() and getShopId(), those are supplied as default.

I worry about wether you should get rid of the permission check of "guest" and "anonymous".  I suspect this will have unexpected consequences in the `order/completed` page, and if you register a user in the Accounts UI-> not being in the checkout flow.

Reply:
I had to remove it because when clicking to continue as guest, the user still has the anonymous permission. So if we are checking to ensure that they are not anonymous, it fails. So that's why I just check if they are guest alone.

Reply:
When you click as "continue as guest" you ARE still `anonymous`.  you are only **NOT** anonymous if you have supplied an email, such as when you add an email at the order/completed stage.

Reply:
Yes that is how it is, but we were preventing `anonymous` from "continuing as guest". which is why I removed the permission for `anonymous`. I will retest and comment guest vs anonymous login view behavior as requested.